relay-general/src/store/normalize.rs

```diff
@@ -273,9 +274,22 @@ fn remove_invalid_measurements(
             Some(m) => m,
             None => return false,
         };
+
+        if !is_valid_metric_name(name) {
+            return false;
+        }
+
         // TODO(jjbayer): Should we actually normalize the unit into the event?
         let unit = measurement.unit.value().unwrap_or(&MetricUnit::None);
 
+        if let Some(max_name_and_unit_len) = max_name_and_unit_len {
+            let max_name_len = max_name_and_unit_len - unit.to_string().len() as isize;
+
+            if name.len() as isize > max_name_len {
+                return false;
+            }
+        }
+
```

Comment:
not sure how to handle having multiple possible errors.

`meta.add_error` is a vector, while `meta.set_original_value` is a single field. if i push various errors to meta, how can i associate them with the correct `original_value`?

Reply:
maybe i could rename `removed_measurements` to `excessive measurements`, and use `original_value` only for the ones removed because they were too many, while for the invalid_names ones i can provide more detailed error messages like `"metric name: '{}' contains invalid characters"` and `"metric name too long. {}/{}"